### PR TITLE
Mean absolute error returns float

### PR DESCRIPTION
**Pull request by willfrey**
Fixes #4961

Changes proposed in this pull request:

- Ensure that `MeanAbsoluteError.get_metric(...)` returns a float, not a tensor.
- Update tests accordingly.

### CODE REVIEW

**Comment by willfrey** [comment]
Letting CI run the tests again to see where I need to throw in a `pytest.approx`

File: allennlp/training/metrics/mean_absolute_error.py [hunk 1]
```diff
@@ -66,9 +66,10 @@ def get_metric(self, reset: bool = False):
         mean_absolute_error = self._absolute_error / self._total_count
         if reset:
             self.reset()
+        assert isinstance(mean_absolute_error, float)
```
Comment (epwalsh) [inline allennlp/training/metrics/mean_absolute_error.py:69]: Since we have this assertion covered in the tests, let's remove it here.

**Review by epwalsh — COMMENTED** [review]
Thanks @willfrey! Just one comment

File: allennlp/training/metrics/mean_absolute_error.py (hunk 1 above)
Comment (willfrey) [reply allennlp/training/metrics/mean_absolute_error.py:69]: Ah, sorry! That was a sanity check for me. I'll remove it.

**Comment by epwalsh** [comment]
Great! Could you just update the CHANGELOG and then we get this merged. You could just add something under the "Fixed" section like "- Ensured `MeanAbsoluteError` metric always returns a `float` instead of a `Tensor`".

**Comment by willfrey** [comment]
@epwalsh Does this look good?

**Review by epwalsh — APPROVED** [review]
LGTM!

P.S. I appreciate your use of emojis in commit messages 👏

**Pull request merged**